x/committee/types/msg.go

```diff
@@ -62,9 +62,10 @@ func (msg MsgSubmitProposal) GetSigners() []sdk.AccAddress {
 type VoteType int
```

Comment:
sdk gotcha: It's best to avoid `int` or `uint` types in state machine code. Instead use one that specifies the size, such as `int64`,`uint8`, `uint64`.

Reason is that for `int` or `uint` types, golang picks a size depending on the hardware and/or OS it's running on. Since validators have different computers, they could get different results for calculations and a different state hash after a block, which leads to a chain fork.

Reply:
@karzak @rhuairahrighairidh we need to compile a list of known Cosmos SDK gotchas (e.g. only iterate over sorted maps) as a reference because one of these days something like this is going to get into the codebase - especially as we onboard additional engineers that may not be as familiar with the Cosmos SDK.

I'll take the first pass at this today and put together a working doc on Confluence.

Reply:
Updated in https://github.com/Kava-Labs/kava/pull/902/commits/23bba6964e52dd2095212ade22c4a7adac3f69a1